Objects/unicodeobject.c

```diff
@@ -5114,9 +5072,14 @@ find_first_nonascii(const unsigned char *start, const unsigned char *end)
             p += SIZEOF_SIZE_T;
         }
     }
+
+    // less than size_t bytes left.
+    assert(end - p <= SIZEOF_SIZE_T);
```

Comment:
This assertion would fail when not `PY_LITTLE_ENDIAN && HAVE_CTZ` case.

```
                // big endian and minor compilers are difficult to test.
                // fallback to per byte check.
                break;
```